Configuration cleanup with tokens processing

lgromanowski opened the pull request:
Hi,
in this pull request there are changes related to configuration clean-up (with tokens). There is also one revert on the top of this branch because I have accidentally pushed workaround for ogre 1.8 to config branch.

When user pass --data command line parameter with some token i.e. ?local:data?/foo/bar/baz then it will be assigned to proper member variable in FixedPath class, and it could be retrieved later by using get*DataPath() ot getInstallPath() methods.

zinnschlag commented:
Sorry, I am having a hard time to understand this code. First the configuration manager header still has those get functions for data paths. These need to go, since (as I explained before) there are no specific data paths for global/user/local locations (well there is data-local, but that has nothing to do with the local location; the name is a bit misleading). The getInstallPath function looks a bit redundant here, though exposing it doesn't do no harm.

It seems you have decided to not to incorporate the data path handling into the configuration manager and instead to stick with the existing system and to run the processPaths functions on the data dir collection. That is okay. Probably even better than what I had originally in mind (less changes), though it makes the name of the configuration manager a bit misleading. But okay.

The biggest problem I see in the implementation of processPaths. If I read it correctly, it works backwards. Instead of adjusting the data paths by replacing the tokens with the locations they represent, it sets the locations associated with these tokens. That can't work (and is also redundant, since these locations are already determined by FixedPath).

zinnschlag commented:
One additional point: I would like to wait with the merging until the whole config task complex is completed, i.e. we have a fully working configuration system. This includes improved openmw.cfg files and the support for reading MW-resources from multiple data paths (the other task I mentioned on the forum).

lgromanowski committed:
Merge pull request #3 from pvdk/config

Changed the order of preference for the configuration files

lgromanowski committed:
Issue #133 Handle resources across multiple data directories

Signed-off-by: Lukasz Gromanowski <lgromanowski@gmail.com>

lgromanowski committed:
Issue #168 - Configuration cleanup

Fixed bug with configuration tokens parsing - when something
appear after token it should be appended to replaced path.

Signed-off-by: Lukasz Gromanowski <lgromanowski@gmail.com>

lgromanowski committed:
Issue #168 - Configuration cleanup

Removed unnecessary path methods - according to forum disscusion:

http://openmw.org/forum/viewtopic.php?f=6&t=448

Signed-off-by: Lukasz Gromanowski <lgromanowski@gmail.com>

lgromanowski commented:
Hi,
if you have some time would you check my recent commits from this pull request please?
